homeassistant/components/devolo_home_control/binary_sensor.py

```diff
@@ -52,6 +67,15 @@ def __init__(self, homecontrol, device_instance, element_uid):
             self._unique_id
         )
 
+        self._device_class = DEVICE_CLASS_MAPPING.get(
+            self._binary_sensor_property.sub_type
+        )
+
+        if self._binary_sensor_property.sub_type == "":
+            self._device_class = DEVICE_CLASS_MAPPING.get(
+                self._binary_sensor_property.sensor_type
+            )
```

Comment:
Maybe something like this?

```suggestion
        self._device_class = DEVICE_CLASS_MAPPING.get(
            self._binary_sensor_property.sub_type
            or self._binary_sensor_property.sensor_type
        )
```

Reply:
@bdraco Changed in latest commit